zebrad/src/commands/start.rs

```diff
@@ -54,6 +54,27 @@ impl StartCmd {
             config.network.network,
         ));
 
+        info!("starting legacy chain check");
+        match state
+            .clone()
+            .oneshot(zebra_state::Request::IsLegacyChain)
+            .await
+        {
+            Ok(zebra_state::Response::LegacyChain(response)) => {
+                if let Some(corrupt_db_path) = response {
+                    // we found a legacy chain
+                    panic!(
+                        "Legacy chain found, database can be corrupted.
+                        Delete your database and retry a full sync.
+                        Database path: {:?}",
+                        corrupt_db_path,
+                    );
```

Comment:
_Panic message changes can be done after the next review. Bug and test changes are more important._

Usability: Let's also tell users the block height and network upgrade that failed. If we return `error_block_height` and  `error_network_upgrade` from `legacy_chain_check`, we can also make the tests more specific. You'll need to add some fields to the `check_transaction_network_upgrade_consistency` error.

```rust
                if let Some(legacy_db_path) = response {
                    panic!(
                        "Cached state contains a legacy chain at {:?}. \
                        An outdated Zebra version did not know about the {:?} network upgrade, \
                        so it followed a legacy chain using {:?} rules. \
                        Hint: Delete your database, and restart Zebra to do a full sync. \
                        Database path: {:?}",
                        error.block_height,
                        error.expected_network_upgrade,
                        error.invalid_network_upgrade,
                        legacy_db_path,
                    );
```